### DISCUSSION TITLE
How to set fs.inotify.max_user_watches permanently for the VM

### DISCUSSION THREAD

**Discussion by ghost** [rancher-sandbox/rancher-desktop#1567] [Q&A] [answered]
Hi!

I am running into some issues with Jest and hot reloading not working because the limit is set to 8192, and that looks to be very low.

I can go manually, ssh into the VM and change that value, but when I restart Rancher Desktop that value gets reset to 8192. 

If I `cat /proc/sys/fs/inotify/max_user_watches` I get 524288, but I still need to run `sudo sysctl -p` to "load" it.

Is there a way of having this value to be permanent?

**Comment by jandubois** [comment, marked as answer]
If you are on macOS or Linux, take a look at https://github.com/rancher-sandbox/rancher-desktop/issues/713#issuecomment-1009377616

**Reply by hendricksond** [reply]
Is there a new way to do this? In rancher desktop 1.13.1 on macos, this doesn't appear to be working:

In `rdctl shell`:
```
$ sysctl fs.inotify
fs.inotify.max_queued_events = 16384
fs.inotify.max_user_instances = 128
fs.inotify.max_user_watches = 61842
```

MacOS:
```
% cat ~/Library/Application\ Support/rancher-desktop/lima/_config/override.yaml 
provision:
- mode: system
  script: |
    #!/bin/sh
    cat <<'EOF' > /etc/security/limits.d/rancher-desktop.conf
    * soft     nofile         524288
    * hard     nofile         524288
    EOF
```

Not sure if it matters, but I am using the dockerd (moby) container engine.

**Reply by AndrewGrossman** [reply]
+1 to @hendricksond's question. I'm encountering this as well and the suggested fix does not appear to be working any longer.

**Reply by StingRayZA** [reply]
This worked for me (rancher-desktop 1.15.1)
@AndrewGrossman @hendricksond 
```
provision:
- mode: system
  script: |
    #!/bin/sh
    echo "fs.inotify.max_user_watches=1048576" > /etc/sysctl.d/fs.inotify.conf
    sysctl -p /etc/sysctl.d/fs.inotify.conf
```

**Reply by AndrewGrossman** [reply]
@StingRayZA's solution appears to work for me.

**Comment by MikeSpreitzer** [comment]
Since the override is applied on every startup of the VM, I have success with the simpler approach of just making the adjustment directly.

```console
mspreitz@mjs13 ~ % cat ~/Library/Application\ Support/rancher-desktop/lima/_config/override.yaml 
provision:
- mode: system
  script: |
    #!/bin/sh
    sysctl fs.inotify.max_user_watches=524288
    sysctl fs.inotify.max_user_instances=512
```

**Comment by GetCurious** [comment]
is there a solution for window? my VM restarts every night 😒
the config persist but everyday i need to run `sudo sysctl -p`

**Reply by StingRayZA** [reply]
You could try this: https://docs.rancherdesktop.io/how-to-guides/increasing-open-file-limit/#windows-steps or this: https://docs.rancherdesktop.io/how-to-guides/provisioning-scripts/#windows

I'm not on windows, so can't speak for it, so YMMV
